data/UnitOfWork.js

```diff
@@ -0,0 +1,27 @@
+const TaskMemoryDAO = require('./TaskMemoryDAO');
+const TaskRepository = require('./TaskRepository');
+const CourseMemoryDAO = require('./CourseMemoryDAO');
+const CourseRepository = require('./CourseRepository');
+
+const validateTaskTitle = (title) => (title && (typeof title === 'string' || title instanceof String));
+const validateCourseId = (id) => (id && Number.isInteger(id));
+
+class UnitOfWork {
+  constructor() {
+    this.taskRepo = new TaskRepository(new TaskMemoryDAO());
+    this.courseRepo = new CourseRepository(new CourseMemoryDAO());
+  }
+
+  createTask(title, courseId) {
+    if (!validateTaskTitle(title) || !validateCourseId(courseId)) {
+      throw new Error('A task must have a valid title and courseId');
+    }
+    const course = this.courseRepo.getById(courseId);
+    if (!course) {
+      throw new Error('A course with the given courseId does not exist');
```

Comment:
If this happens then the error message will be clues for the dev in debugging so it might be good to print out the id

```javascript
throw new Error(`Course with id ${courseId} does not exist`);
```